### ISSUE TITLE
Bug with loading of extension in IPython

### ISSUE THREAD

**Issue by jorisvandenbossche** [micheles/decorator#105]
With the 5.0 release, I noticed that loading an extension (magic) in IPython started to fail:

```
In [1]: %load_ext cython
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-1-a0787ba4fffb> in <module>
----> 1 get_ipython().run_line_magic('load_ext', 'cython')

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/IPython/core/interactiveshell.py in run_line_magic(self, magic_name, line, _stack_depth)
   2342                 kwargs['local_ns'] = self.get_local_scope(stack_depth)
   2343             with self.builtin_trap:
-> 2344                 result = fn(*args, **kwargs)
   2345             return result
   2346 

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/decorator.py in fun(*args, **kw)
    212     else:
    213         def fun(*args, **kw):
--> 214             return caller(func, *(extras + args), **kw)
    215     fun.__name__ = func.__name__
    216     fun.__signature__ = inspect.signature(func)

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/IPython/core/magic.py in <lambda>(f, *a, **k)
    185     # but it's overkill for just that one bit of state.
    186     def magic_deco(arg):
--> 187         call = lambda f, *a, **k: f(*a, **k)
    188 
    189         if callable(arg):

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/IPython/core/magics/extension.py in load_ext(self, module_str)
     31         if not module_str:
     32             raise UsageError('Missing module name.')
---> 33         res = self.shell.extension_manager.load_extension(module_str)
     34 
     35         if res == 'already loaded':

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/IPython/core/extensions.py in load_extension(self, module_str)
     85                               dir=compress_user(self.ipython_extension_dir)))
     86             mod = sys.modules[module_str]
---> 87             if self._call_load_ipython_extension(mod):
     88                 self.loaded.add(module_str)
     89             else:

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/IPython/core/extensions.py in _call_load_ipython_extension(self, mod)
    132     def _call_load_ipython_extension(self, mod):
    133         if hasattr(mod, 'load_ipython_extension'):
--> 134             mod.load_ipython_extension(self.shell)
    135             return True
    136 

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/Cython/__init__.py in load_ipython_extension(ip)
      9 def load_ipython_extension(ip):
     10     """Load the extension in IPython."""
---> 11     from .Build.IpythonMagic import CythonMagics  # pylint: disable=cyclic-import
     12     ip.register_magics(CythonMagics)

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/Cython/Build/IpythonMagic.py in <module>
    561     CYTHON_DOC=dedent(CythonMagics.cython.__doc__\
    562                                   .replace('-+, --cplus', '--cplus    ')),
--> 563     CYTHON_INLINE_DOC=dedent(CythonMagics.cython_inline.__doc__),
    564     CYTHON_PYXIMPORT_DOC=dedent(CythonMagics.cython_pyximport.__doc__),
    565 )

~/miniconda3/envs/test-decorator/lib/python3.9/site-packages/IPython/utils/text.py in dedent(text)
    368     """
    369 
--> 370     if text.startswith('\n'):
    371         # text starts with blank line, don't ignore the first line
    372         return textwrap.dedent(text)

AttributeError: 'NoneType' object has no attribute 'startswith'
```

I don't know if it's directly caused by `decorator`, but at least it started to fail with the `decorator` version bump (and no other package version changed). 
Noticed this with the Cython magic, didn't yet check if it's with others as well.

**Comment by jorisvandenbossche** [comment]
It seems specific with the Cython magic. It accesses `CythonMagics.cython_inline.__doc__)`, which previously returned a string but now returns None. And the ``cython_inline`` method is a decorated method, with a decorator constructed using `decorator`.

I don't fully understand the code, but the IPython magic decorator is defined here: https://github.com/ipython/ipython/blob/657384935a24be05f2725d41ea0eac271863e6d1/IPython/core/magic.py#L178-L209

So not fully clear to me where the issue is (in decorator, IPython or cython)

**Comment by micheles** [comment]
This is fixed now, just upgrade to release 5.0.4 and thanks for the bug report.

**Comment by jorisvandenbossche** [comment]
Thanks for the quick fix!
